src/Piece.js

```diff
@@ -1,5 +1,47 @@
 /* Represents a piece in the game */
 class Piece 
 {
+	constructor()
+	{
+		this.moveVectors = [];
+		this.captureVectors = [];
+		this.moveCaptureVectors = [];
+		this.player = undefined;
+	}
 	
-}
\ No newline at end of file
+	static Create(baseObj)
```

Comment:
Why do you have what is essentially a constructor in this `Create` method?

Reply:
These are factory methods. Although we really don't need them right now, and I don't know if we ever will want to collapse / separate functionality into subclasses (which is the point of a factory method). Can easily be changed to a constructor if we want to.